include/fiction/io/write_sidb_layout_to_svg_in_siqad_style.hpp

```diff
@@ -0,0 +1,417 @@
+            fmt::format(R"(<rect x="{0}" y="{1}" width="{2}" height="{3}" style="fill:{4};"/>)", viewbox_x, viewbox_y,
+                        viewbox_width, viewbox_height, background_color);
+
+        // Generate the final SVG content
+        os << fmt::format(fiction::svg::HEADER_TEMPLATE, FICTION_VERSION, FICTION_REPO, viewbox_x, viewbox_y,
+                          viewbox_width, viewbox_height, formatted_path_definition, background_rect, svg_content.str());
+    }
+};
+
+}  // namespace detail
+
+/**
+ * Writes an SVG representation of a cell-level SiDB layout into an output stream.
+ *
+ * @tparam Lyt SiDB cell-level layout type.
+ * @param lyt The layout to be written.
+ * @param os The output stream to write into.
+ * @param ps Parameters.
+ *
+ * @note SiDB defects are not supported yet.
+ */
+template <typename Lyt>
+void write_sidb_layout_to_svg_in_siqad_style(const Lyt& lyt, std::ostream& os,
+                                             const write_sidb_layout_svg_in_siqad_style_params& ps = {})
+{
+    static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
+    static_assert(has_sidb_technology_v<Lyt>, "Lyt must be a SiDB layout");
+    static_assert(!is_sidb_defect_surface_v<Lyt>, "SiDB defects are not supported");
+
+    detail::write_sidb_layout_svg_impl<Lyt> p{lyt, os, ps};
+    p.run();
+}
+
+/**
+ * Writes an SVG representation of a cell-level SiDB layout into a file.
+ *
+ * @tparam Lyt SiDB cell-level layout type.
+ * @param lyt The layout to be written.
+ * @param filename The file name to create and write into.
+ * @param ps Parameters.
+ *
+ * @note SiDB defects are not supported yet.
+ */
+template <typename Lyt>
+void write_sidb_layout_to_svg_in_siqad_style(const Lyt& lyt, const std::string_view& filename,
+                                             const write_sidb_layout_svg_in_siqad_style_params& ps = {})
+{
+    static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
+    static_assert(has_sidb_technology_v<Lyt>, "Lyt must be a SiDB layout");
+    static_assert(!is_sidb_defect_surface_v<Lyt>, "SiDB defects are not supported");
+
+    std::ofstream os{filename.data(), std::ofstream::out};
+
+    if (!os.is_open())
+    {
+        throw std::ofstream::failure("Could not open file");
+    }
+
+    write_sidb_layout_to_svg_in_siqad_style(lyt, os, ps);
+    os.close();
+}
+/**
+ * This function creates an SVG representation of the given SiDB layout and returns the
+ * result as a string. The SVG is generated according to SiQAD style parameters and is
+ * suitable for visualization.
+ *
+ * @tparam Lyt SiDB cell-level layout type.
+ * @param lyt The input layout to be converted to an SVG string.
+ * @param ps Optional parameters to customize the SVG generation process.
+ *           Defaults to an instance of `write_sidb_layout_svg_in_siqad_style_params`.
+ * @return A string containing the SVG representation of the provided SiDB layout.
+ *
+ */
```

Comment:
Why does this docstring differ from the others in style and content? Also, the note is missing.